### ISSUE TITLE
Receiving trade request from hidden player crashes ClassicUO client

### ISSUE THREAD

**Issue by canterberry** [ClassicUO/ClassicUO#728]
**Describe the bug**

ClassicUO crashes when receiving a secure trade request from a hidden player.
This can be exploited to crash the client of any ClassicUO user.

**To Reproduce**

 1. Ask any other player to hide then drag any item onto your character to initiate a trade window.
 2. Crash 💥 

**Expected behavior**

No effect. The initiator of the trade is hidden, so no trade should be able to occur.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: Commit 7ce33011 (latest dev as of the creation of this issue)

```
ClassicUO [dev] - v0.1.2.1
OS: Win32NT x64
Thread: CUO_MAIN_THREAD

Exception:
System.NullReferenceException: Object reference not set to an instance of an object.
   at ClassicUO.Network.PacketHandlers.UpdateContainedItem(Packet p) in C:\path\to\ClassicUO\src\Network\PacketHandlers.cs:line 1053
   at ClassicUO.Network.PacketHandlers.OnPacket(Object sender, Packet p) in C:\path\to\ClassicUO\src\Network\PacketHandlers.cs:line 79
   at ClassicUO.Utility.Exstentions.Raise[T](EventHandler`1 handler, T e, Object sender) in C:\path\to\ClassicUO\src\Utility\Exstentions.cs:line 47
   at ClassicUO.Network.NetClient.Update() in C:\path\to\ClassicUO\src\Network\NetClient.cs:line 235
   at ClassicUO.Engine.OnNetworkUpdate(Double totalMS, Double frameMS) in C:\path\to\ClassicUO\src\Engine.cs:line 803
   at ClassicUO.Engine.Tick() in C:\path\to\ClassicUO\src\Engine.cs:line 667
   at Microsoft.Xna.Framework.SDL2_FNAPlatform.RunLoop(Game game)
   at Microsoft.Xna.Framework.Game.Run()
   at ClassicUO.Engine.Run(String[] args) in C:\path\to\ClassicUO\src\Engine.cs:line 323
   at ClassicUO.Bootstrap.Main(String[] args) in C:\path\to\ClassicUO\src\Main.cs:line 35
```

**Comment by canterberry** [comment]
There are a couple of root causes for this issue:

 1. When receiving an `UpdateContainedItem` packet, there is no defensive `null` check for whether the mobile identified by the `containerSerial` actually exists in the world. There needs to be a defensive check here, else a malicious packet would be able to crash the client by sending a packet with a dubious container serial.

 2. The hidden player character is not in ClassicUO's list of mobiles. Should it be?

After adding a fix for (1) locally, I'm able to successfully complete a trade from a hidden player to an unhidden player. However, because the `UpdateContainedItem` packet is not processed, the traded item(s) do not actually appear in the trade window (though the trade *is* completed successfully and the items end up in the appropriate packs).

It appears that the server may be configured to allow trades between a hidden player and another player, therefore ClassicUO should support this as well. Therefore, there must be some way of successfully handling the `UpdateContainedItem` packet, even if the initiating mobile is not in ClassicUO's mobile list.

**Comment by andreakarasho** [comment]
The hidden mobile is deleted by the server so the dictionary will not have that mobile.
A null check is enough I guess

**Comment by Stallind** [comment]
A null check is not enough. Im trying to solve it

**Comment by Stallind** [comment]
Or actually it is, if the server supports it. Im surprised im handing items from an invis char now. lol

**Comment by canterberry** [comment]
@Stallind In case it helps, here's a few other things I've tried:

 1. Attempt to find the `TradingGump`, if any, in the UI engine, then attempt to update the container(s) as identified in the gump's `ID1`, `ID2`, and/or `LocalSerial`.

 2. Attempt to find the "secure box" item via the `World.Player` instead of via the (in this case, non-existent) mobile identified by the packet's given `containerSerial`.

I haven't had success, yet, getting the traded items to appear in the trade gump, although as noted, I am able to see those trades go through successfully and items transferred as expected.

As noted earlier, with a simple `null` check, I'm able to mitigate the crash issue. This may be an adequate **short-term** fix so that players can no longer crash other players. ~~I'll put together a quick PR for this, at least.~~ (edit: Stallind's got it covered in PR #729)

**Comment by canterberry** [comment]
Fixed in #729.
Will create a follow-up for the remaining issue, which is much lower priority to fix.
